### DISCUSSION TITLE
Bypassing the URL shortener

### DISCUSSION THREAD

**Discussion by benjaminleighton** [TerriaJS/terriajs#6848] [Q&A] [answered]
By default opening the "share" button sends data to the URL shortener. As I understand it local data doesn't get sent but metadata about user added data services will be sent. In some cases this might not be desirable for example if I'm accessing "http://my.intranet/some_sensitive_project_name/data_service". Is there a way through the start parameter and an init fragement to disable this? Otherwise is it worth opening an issue?

**Comment by na9da** [comment]
Hi @benjaminleighton,

The only two options I can think of right now are:

1) If you are the map owner and can modify the server configuration, then you can disable the URL-shortening service for the entire map by removing or setting `newShareUrlPrefix` option to `null`. You can check your current config settings by loading `example.com/serverconfig`. Here is an [example config](http://ci.terria.io/main/serverconfig) for our CI server - `newShareUrlPrefix` is set to `s`.

2) The other option is to disable the URL shortener  from the UI. The app remembers your selection and so will not call the shortener unless you enable it again. 
   ![image](https://github.com/TerriaJS/terriajs/assets/1430/349b60ff-4e7b-442f-8598-8b5fa68034ba)

Are these workable in your case? Otherwise we might have to consider adding a URL option like `disableUrlShortener=1` (similar to the [hideWorkbench](http://ci.terria.io/main/#hideWorkbench=1) option)

**Reply by benjaminleighton** [reply]
Thanks for your quick response. We want to provide an "Open in National Map" button in another application and use nationalmap.gov.au. Likely neither option would be workable for us, I'm not the map owner and users are unlikely to have opened National Map prior and disabled the Shorten option and it would be unreliable to ask them to do that. A disableUrlShortener option would be great I think! It looks like some useful logic might exist at https://github.com/TerriaJS/terriajs/blob/b14835b75232eb8be722c6255f9c6e1789b1f5cb/lib/ReactViews/Map/Panels/SharePanel/BuildShareLink.ts#L252

**Reply by na9da** [reply, marked as answer]
That sounds like a valid use case @benjaminleighton. 
I have created a new [issue here](https://github.com/TerriaJS/terriajs/issues/6852).
